algoliasearch-core/src/main/java/com/algolia/search/SearchClientAPIKeys.java

```diff
@@ -422,4 +426,35 @@ default String generateSecuredAPIKey(
       @Nonnull String parentAPIKey, SecuredApiKeyRestriction restriction) throws Exception {
     return HmacShaUtils.generateSecuredApiKey(parentAPIKey, restriction);
   }
+
+  /**
+   * Gets how many seconds are left before the secured API key expires.
+   *
+   * @param securedAPIKey The secured API Key to check
+   * @throws AlgoliaRuntimeException if <code>securedAPIKey</code> is null, empty or whitespaces.
+   * @throws AlgoliaRuntimeException if <code>securedAPIKey</code> doesn't have a <code>validUntil
+   *     </code> parameter.
+   */
+  default Duration getSecuredApiKeyRemainingValidity(@Nonnull String securedAPIKey) {
+
+    if (AlgoliaUtils.isNullOrEmptyWhiteSpace(securedAPIKey)) {
+      throw new AlgoliaRuntimeException("securedAPIKey must not be empty, null or whitespaces");
+    }
+
+    byte[] decodedBytes = Base64.getDecoder().decode(securedAPIKey);
+    String decodedString = new String(decodedBytes);
+
+    Pattern pattern = Pattern.compile("validUntil=\\d{10}");
```

Comment:
Why is there a 10 digits limit? Can't we use `\\d+` instead?

Reply:
I choose `\\d{10}` because the UNIX timestamp returned by the engine will have 10 digits length! Do you think it's too restrictive and not future proof?